### PR TITLE
Add installation guide

### PR DESCRIPTION
**Pull request by stefanprodan**
This PR adds the toolkit installation guide covering repository bootstrap for GitHub and GitLab, manual bootstrap and dev install options.

### CODE REVIEW

File: docs/guides/installation.md [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+If the toolkit components are present on the cluster,
+the bootstrap command will perform an upgrade if needed.
+The bootstrap is idempotent, it's safe to run the command as many times as you want.
+
+You can choose what components to install and for which cluster with:
+
+```sh
+tk bootstrap <GIT-PROVIDER> \
+  --components=source-controller,kustomize-controller,helm-controller,notification-controller \
+  --path=my-cluster \
```
Comment (dholbach) [inline docs/guides/installation.md:46]: Might it be worth explaining this `--path` argument?

File: docs/guides/installation.md [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+tk bootstrap gitlab \
+  --hostname=my-gitlab.com \
+  --owner=my-gitlab-group \
+  --repository=my-repository \
+  --path=my-cluster
+```
+
+### Generic Git Server
+
+For other Git providers such as Bitbucket, Gogs, Gitea, etc you can manually setup the repository and the deploy key.
```
Comment (dholbach) [inline docs/guides/installation.md:139]: Maybe say a word about talking to us if anyone was interested in contributing support for one of these git providers, that we'd happily add it to the roadmap?

**Comment by dholbach** [comment]
Maybe link this article from the get-started document too (e.g. "if you need more detailed information on installing the toolkit or have a different setup, ...")?

**Comment by stefanprodan** [comment]
> Maybe link this article from the get-started document 

If you look at the footer, you'll see that installation is the next one proposed.

File: docs/guides/installation.md (hunk 2 above)
Comment (stefanprodan) [reply docs/guides/installation.md:153]: I would wait for the bootstrap to be [ported](https://github.com/fluxcd/go-git-providers/issues/11) over to go-git-providers than add that repo here.

Comment (dholbach) [reply docs/guides/installation.md:153]: Ah yes, good thinking!

**Pull request merged**